pandas/core/ops/__init__.py

```diff
@@ -324,7 +322,10 @@ def dispatch_to_series(left, right, func, str_rep=None, axis=None):
         return type(left)(bm)
 
     elif isinstance(right, ABCDataFrame):
-        assert right._indexed_same(left)
+        assert left.index.equals(right.index)
+        assert left.columns.equals(right.columns)
+        # TODO: The previous assertion `assert right._indexed_same(left)`
+        #  fails in cases with shaoe[1] == 0
```

Comment:
I don't really understand this comment

Reply:
im finding cases where left.columns and right.columns are both empty but different dtypes, which still satisfy `left.columns.equals(right.columns)`, but not `right._indexed_same(left)`.  looks like this is an artifact of the intersection done in `_should_reindex_frame_op`, looking into this now.  (also will fix the typo shaoe -> shape)